clang/test/Sema/main-no-return-c89.c

```diff
@@ -0,0 +1,6 @@
+/* RUN: %clang_cc1 -std=c89 -fsyntax-only -verify -pedantic -Wno-strict-prototypes %s
```

Comment:
Can you also add RUN lines for: `-Wmain-return-type` (enables diagnostic), `-pedantic -Wno-main-return-type` (disables diagnostic), and `` (disables diagnostic)?

Reply:
@AaronBallman I've added additonal tests